### DISCUSSION TITLE
Elegant solution for using .dat files?

### DISCUSSION THREAD

**Discussion by KareemKarl** [coin-or/pulp#366] [Q&A]
Hey guys,

I just wanted to ask if there was an easy way to use .dat files in PuLP?

I am familiar with the glpsol method of passing the .dat file as an argument in the command line. However, I am new to PuLP and have not yet figured out if there is a simple way to incorporate .dat files. 

Is the only solution to use numpy.genfromtext() ?

Thanks in advance!

**Comment by pchtsp** [comment]
I have no idea what the .dat file is, sorry.
Pulp can read mps files and json files: https://coin-or.github.io/pulp/guides/how_to_export_models.html

**Comment by tkralphs** [comment]
I assume that @KareemKarl is referring to files in the format that AMPL uses for specifying problem data. [Amply](https://github.com/willu47/amply) is probably what you want. It used to be part of PuLP, actually, right? [Pyomo](https://github.com/Pyomo/pyomo) can also read AMPL data, but then you have to use Pyomo for building the model.
